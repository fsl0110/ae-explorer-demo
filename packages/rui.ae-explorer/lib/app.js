"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var app_styles_1 = require("./app.styles");
var components_1 = require("./components");
var groups_1 = require("./pages/groups/groups");
var details_redux_1 = require("./connections/details/details.redux");
exports.App = function (_a) {
    var dataProp = _a.data;
    var _b = react_1.useState(dataProp), data = _b[0], setData = _b[1];
    react_1.useEffect(function () {
        setData(dataProp);
    }, [dataProp]);
    var groups = function () { return react_1.default.createElement(groups_1.Groups, { data: data }); };
    var details = function () { return react_1.default.createElement(details_redux_1.DetailsConnection, { data: data }); };
    var arr = window.location.pathname.split("/");
    return (react_1.default.createElement(app_styles_1.AppStyles, null,
        react_1.default.createElement(components_1.Header, { data: data }),
        react_1.default.createElement(react_router_dom_1.Switch, null,
            react_1.default.createElement(react_router_dom_1.Route, { exact: true, path: "/" + arr[1], component: groups }),
            react_1.default.createElement(react_router_dom_1.Route, { path: "/" + arr[1] + "/:id", component: details }))));
};
//# sourceMappingURL=app.js.map