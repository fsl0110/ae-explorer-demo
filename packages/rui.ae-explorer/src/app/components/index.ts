export {GroupsSelections} from "./groupsSelections/groupsSelections";
export {Header} from "../components/header/header";
export {GroupsTable} from "../components/groupsTable/groupsTable";
export {Filters} from "./filters/filters";
export {GroupsTableManager} from "./groupsTableManager/groupsTableManager";
export {Share} from "./share/share";
