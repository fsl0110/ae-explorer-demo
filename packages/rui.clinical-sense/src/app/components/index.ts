export {Counter} from "./counter/counter";
export {Header} from "./header/header";
