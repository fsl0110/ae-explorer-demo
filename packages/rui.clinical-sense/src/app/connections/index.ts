export {CounterRedux} from "./counterConnection/counter.redux";
export {CounterRedux as CounterConnect} from "./counterConnection/counter.redux";
export {CounterSocket} from "./counterConnection/counter.socket";
