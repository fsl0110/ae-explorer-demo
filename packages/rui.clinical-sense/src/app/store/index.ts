import {createStore, combineReducers} from "redux";
import {composeWithDevTools} from "redux-devtools-extension";
import {persistStore} from "redux-persist";
import {counterReducer, counterState, CounterState} from "./reducers/counter.reducers";
import {formReducer} from "./reducers/form.reducers";

const rootReducer = combineReducers({counter: counterReducer, form: formReducer});

export interface AppState {
  counter: CounterState;
}

export const initialState: AppState = {
  counter: counterState,
};

export const store = createStore(rootReducer, initialState, composeWithDevTools());

export const persistor = persistStore(store);
