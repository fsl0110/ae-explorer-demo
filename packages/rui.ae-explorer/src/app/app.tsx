import React, {FC, useState, useEffect} from "react";
import {Route, Switch} from "react-router-dom";
import {AppStyles} from "./app.styles";
import {Header} from "./components";
import {Groups} from "./pages/groups/groups";
import {DetailsConnection} from "./connections/details/details.redux";
import {Data} from "./types";

interface Props {
  data: Data[];
}

export const App: FC<Props> = ({data: dataProp}) => {
  const [data, setData] = useState<Data[]>(dataProp);

  useEffect(() => {
    setData(dataProp);
  }, [dataProp]);

  const groups = () => <Groups data={data} />;
  const details = () => <DetailsConnection data={data} />;

  const arr = window.location.pathname.split("/");

  return (
    <AppStyles>
      <Header data={data} />
      <Switch>
        <Route exact={true} path={`/${arr[1]}`} component={groups} />
        <Route path={`/${arr[1]}/:id`} component={details} />
      </Switch>
    </AppStyles>
  );
};
