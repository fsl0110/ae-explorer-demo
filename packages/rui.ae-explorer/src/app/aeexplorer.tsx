import React, {FC} from "react";
import {Provider} from "react-redux";
import {PersistGate} from "redux-persist/lib/integration/react";
import {BrowserRouter as Router} from "react-router-dom";
import {store, persistor} from "./store";
import {Data} from "./types";
import {App} from "./app";

export interface Props {
  data: Data[];
}

export const AEExplorer: FC<Props> = ({data}) => (
  <Provider store={store}>
    <PersistGate persistor={persistor}>
      <Router>
        <App data={data} />
      </Router>
    </PersistGate>
  </Provider>
);
