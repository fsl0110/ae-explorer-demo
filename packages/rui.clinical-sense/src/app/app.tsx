import React, {FC, useState} from "react";
import {Provider} from "react-redux";
import {BrowserRouter as Router, Route, Switch, Redirect} from "react-router-dom";
import {store} from "./store";
import {Header} from "./components";
import {AppStyles} from "./app.styles";
import "antd/dist/antd.css";
import {AEExplorer} from "@dedrr/rui.ae-explorer";
import data from "./data.json";
import tinyData from "./data.tiny.json";

export const App: FC = () => {
  const [aeData, setAeData] = useState(data);

  const handleDataChange = (payload: string) => {
    const nextData = payload === "full" ? data : tinyData;
    setAeData(nextData);
  };

  const aeExplorer = () => <AEExplorer data={aeData} />;

  const redirect = () => <Redirect to="/aeexplorer" />;

  return (
    <Provider store={store}>
      <Router>
        <AppStyles>
          <Header handleChange={handleDataChange} />
          <Switch>
            <Route exact={true} path="/" render={redirect} />
            <Route path="/aeexplorer" component={aeExplorer} />
          </Switch>
        </AppStyles>
      </Router>
    </Provider>
  );
};
