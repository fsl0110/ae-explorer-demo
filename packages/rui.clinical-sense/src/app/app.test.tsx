import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {App} from "./app";

describe("<App>", () => {
  it("Snapshot", () => {
    expect(toJson(shallow(<App />))).toMatchSnapshot();
  });
});
