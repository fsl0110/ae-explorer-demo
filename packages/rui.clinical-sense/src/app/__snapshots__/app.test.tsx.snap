// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <styled.div>
        <Header />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/contact"
          />
          <Route
            component={[Function]}
            path="*"
          />
        </Switch>
      </styled.div>
    </BrowserRouter>
  </PersistGate>
</Provider>
`;
